src/kernel/include/shared/cos_types.h

```diff
@@ -259,8 +259,22 @@ enum {
 
 enum {
 	BOOT_MEM_VM_BASE = (COS_MEM_COMP_START_VA + (1<<22)), /* @ 1G + 8M */
-	BOOT_MEM_KM_BASE = PAGE_SIZE, /* kernel memory @ first page, not at address 0 to avoid NULL */
-	BOOT_MEM_PM_BASE = 0x80000000, /* user memory @ 2 GB */
+	BOOT_MEM_KM_BASE = PGD_SIZE, /* kernel & user memory @ 4M, pgd aligned start address */
+};
+
+enum {
+	/* thread register states */
+	THD_GET_IP,
+	THD_GET_SP,
+	THD_GET_BP,
```

Comment:
we should likely try and make these generic across architectures.

BP = FP (frame pointer)
IP = IP
SP = SP
AX = 1
BX = 2
CX = 3
DX = 4
SI = 5
DI = 6

(as in `THD_GET_1`)

Another question is if we want these are generic register identifiers, or if we want the action `GET` associated with them.  This is fine for now, but we might change them in the future.